code/modules/clothing/neck/bodycamera.dm

```diff
@@ -0,0 +1,154 @@
+	var/setup = FALSE
+	var/preset = FALSE //if true, the camera is already configured and cannot be reset
+	actions_types = list(/datum/action/item_action/toggle_bodycam)
+	strip_delay = 1 SECONDS //takes one second to strip, so a downed officer can be un-cammed quickly
+	w_class = WEIGHT_CLASS_NORMAL
+	var/mob/listeningTo //This code is simular to the code for the RCL.
+
+/obj/item/clothing/neck/bodycam/Initialize()
+	..()
+	ADD_TRAIT(src, TRAIT_NO_STORAGE, TRAIT_GENERIC)
+	bodcam = new(src)
+	bodcam.c_tag = "NT_BodyCam"
+	bodcam.network = list("ss13")
+	bodcam.internal_light = FALSE
+	bodcam.status = FALSE
+	update_icon()
+
+/obj/item/clothing/neck/bodycam/attack_self(mob/user)
+	if(!setup)
+		AltClick(user)
+		return
+	if(bodcam.status)
+		to_chat(user, "<span class='notice'>You shut off the body camera.</span>")
+		Disconnect()
+	else
+		bodcam.status = TRUE
+		to_chat(user, "<span class='notice'>You turn on the body camera.</span>")
+		getMobhook(user)
+	update_icon()
+
+/obj/item/clothing/neck/bodycam/AltClick(mob/user)
+	if(preset)
+		return //can't change the settings on it if it's preset
+	var/name = stripped_input(user, "What would you like your camera's display name to be?", "Camera id", "[user.name]")
+	if(name)
+		bodcam.c_tag = "(Bodycam) [name]"
+	var/temp = stripped_input(user, "Which network should the camera broadcast to?\nFor example, 'ss13', 'security', and 'mine' are existing networks", "Camera network", "ss13")
+	if(temp)
+		bodcam.network[1] = temp
+		setup = TRUE
+		bodcam.status = TRUE
+		update_icon()
+
+/obj/item/clothing/neck/bodycam/update_icon()
+	..()
+	var/suffix = "off"
+	if (bodcam.status)
+		suffix = "on"
+	icon_state = "[prefix]_bodycam_[suffix]"
+	item_state = "[prefix]_bodycam_[suffix]"
+	for(var/X in actions)
+		var/datum/action/A = X
+		A.UpdateButtonIcon()
+
+/obj/item/clothing/neck/bodycam/examine(mob/user)
+	.=..()
+	. += "<span class='notice'>The camera is currently [bodcam.status ? "on" : "off"].</span>"
+	if(setup)
+		. += "<span class='notice'>It is registered under the name \"[bodcam.c_tag]\".</span>"
+		. += "<span class='notice'>It is streaming to the network \"[bodcam.network[1]]\".</span>"
+		if(!preset)
+			. += "<span class='notice'>Alt-click to configure the camera.</span>"
+		else
+			. += "<span class='notice'>This camera is locked and cannot be reconfigured.</span>"
+	else
+		. += "<span class='warning'>It hasn't been set up yet!</span>"
+
+/obj/item/clothing/neck/bodycam/verb/toggle_bodycam()
+	set name = "Toggle Bodycam"
+	set category = "Object"
+	set src in oview(1)
+
+	if(!usr.incapacitated())
+		attack_self(usr)
+
+/obj/item/clothing/neck/bodycam/emp_act(severity)
+	. = ..()
+	if(prob(150/severity))
+		Disconnect()
+		bodcam.c_tag = rand(1,10000)
+		bodcam.network[1] = rand(1, 10000) //gibberish, this will render the camera basically unreadable by any console
+		update_icon()
+
+/obj/item/clothing/neck/bodycam/Destroy()
+	. = ..()
+	Disconnect()
+	QDEL_NULL(bodcam)
```

Comment:
1) Id really appreciate it if you can keep initilize and destroy together at the top, its helpful to be able to keep track of what gets created and deleted
2) only call ..() AFTER youre done with the destroy for the current object (aka move the . = ..() at the end)